module/zfs/dmu_object.c

```diff
@@ -50,6 +59,9 @@ dmu_object_alloc_dnsize(objset_t *os, dmu_object_type_t ot, int blocksize,
 	dnode_t *dn = NULL;
 	int dn_slots = dnodesize >> DNODE_SHIFT;
 	boolean_t restarted = B_FALSE;
+	uint64_t *cpuobj = &os->os_obj_next_array[CPU_SEQID %
+	    os->os_obj_next_array_len];
+	int dnodes_per_chunk = 1 << dmu_object_alloc_chunk_shift;
```

Comment:
Correct, the cpuobj should typically typically different on different CPUs, but if it happens to be the same that is handled.